cloudweatherreport/datastore.py

```diff
@@ -87,25 +97,60 @@ def lock(self, timeout=5*60):
         # Optimistically create our unique lock file.  This relies on RAW
         # consistency to ensure it will be immediately visible to others,
         # and their lock files to us.
-        lock_id = uuid4()
+        lock_id = self.create_lock_id()
         lock_filename = '.lock.{}'.format(lock_id)
         self.write(lock_filename, '')
+        log.debug("Trying to acquire datastore lock {}".format(lock_filename))
         try:
             # wait until we own the earliest lock file, and thus the lock
             wait_secs = 1
             total_waited = 0
-            while self._active_lock_filename() != lock_filename:
+            active_lock = self._active_lock_filename()
+            while active_lock != lock_filename:
+                log.debug('Lock already acquired {} age:{} sec.'.format(
+                    active_lock, int(self.age_in_seconds(active_lock))))
+                self.delete_old_lock(active_lock, old_lock_age)
                 sleep(wait_secs)
                 total_waited += wait_secs
                 if total_waited >= timeout:
-                    raise TimeoutError('Timed out waiting for lock')
+                    raise TimeoutError(
+                        'Timed out waiting for lock:{}'.format(lock_filename))
                 # increase sleep time a second at a time, up to 10s
                 if wait_secs < 10:
                     wait_secs += 1
+                active_lock = self._active_lock_filename()
+            log.debug('Datastore lock acquired {}'.format(lock_filename))
             yield lock_id
         finally:
+            log.debug('Datastore lock released {}'.format(lock_filename))
             self.delete(lock_filename)
 
+    def create_lock_id(self):
+        """Create lock id.
+
+        If it is running in Jenkins, it will postfix the lock id with the job
+        name and build number.
+        """
+        job_name = os.getenv('JOB_NAME')
+        build_number = os.getenv('BUILD_NUMBER')
+        if job_name and build_number:
+            return '{}.{}.{}'.format(uuid4(), job_name, build_number)
+        return uuid4()
+
+    def delete_old_lock(self, lock, old_age):
```

Comment:
Minor nitpick: I would call this "delete_if_old" or something else to indicate that it might or might not delete the lock, depending on a check. Not a blocker for merging, though.